fakeserver/fakeserver_test.go

```diff
@@ -0,0 +1,152 @@
+package fakeserver
```

Comment:
Should the filename be `fakeserver/server_test.go` to mirror `fakeserver/server.go`? Or is the idea that we're testing the whole package in a more integration-ey fashion?

Reply:
yeah, the latter was the idea. i did a bit of reading to see if there were conventions to follow and found an approach that may apply here. we could introduce a new package `fakeserver_test` which is meant for black box testing of the whole package.

to pull that off, i'd need to expose more of the package internals, the `createHandler` function and the serialization types.

i'd rather not go down that route, so lacking a good option, i'll take your suggestion and rename it so we're at least following some convention.